File: behaviorial/mediator/python/component.py
from mediator import Component


class Checkbox(Component):
    def __init__(self):
        super().__init__()
        self.checked = False

    def toggle(self):
        self.checked = not self.checked
        print(f"Checkbox toggled: {self.checked}")
        self.mediator.notify(self, 'checked')

    def is_checked(self):
        return self.checked


class TextField(Component):
    def __init__(self):
        super().__init__()
        self.value = ''
        self.visible = False

    def set_visible(self, visible: bool):
        self.visible = visible
        print(f"Text field visibility: {self.visible}")

    def set_value(self, value: str):
        self.value = value

    def get_value(self):
        return self.value


class Button(Component):
    def click(self):
        print("Button clicked")
        self.mediator.notify(self, 'click')

File: behaviorial/mediator/typescript/component.ts
import { Mediator } from './mediator';

export abstract class Component {
	protected mediator!: Mediator;

	setMediator(mediator: Mediator): void {
		this.mediator = mediator;
	}
}

export class Checkbox extends Component {
	private checked: boolean = false;

	toggle(): void {
		this.checked = !this.checked;
		console.log(`Checkbox toggled: ${this.checked}`);
		this.mediator.notify(this, 'checked');
	}

	isChecked(): boolean {
		return this.checked;
	}
}

export class TextField extends Component {
	private value: string = '';
	private visible: boolean = false;

	setVisible(visible: boolean): void {
		this.visible = visible;
		console.log(`Text field visibility: ${this.visible}`);
	}

	setValue(value: string): void {
		this.value = value;
	}

	getValue(): string {
		return this.value;
	}
}

export class Button extends Component {
	click(): void {
		console.log('Button clicked');
		this.mediator.notify(this, 'click');
	}
}
